test/PrintAsObjC/mixed-framework.swift

```diff
@@ -1,17 +1,20 @@
-// RUN: %empty-directory(%t)
-// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -F %S/Inputs/ -module-name Mixed -import-underlying-module -parse-as-library %s -typecheck -emit-objc-header-path %t/mixed.h
-// RUN: %FileCheck -check-prefix=CHECK -check-prefix=FRAMEWORK %s < %t/mixed.h
-// RUN: %check-in-clang -F %S/Inputs/ %t/mixed.h
+// RUN: %empty-directory(%t.framework)
+// RUN: %empty-directory(%t.framework/Headers)
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -F %S/Inputs/ -module-name Mixed -import-underlying-module -parse-as-library %s -typecheck -emit-objc-header-path %t.framework/Headers/mixed.h
+// RUN: %FileCheck -check-prefix=CHECK -check-prefix=FRAMEWORK %s < %t.framework/Headers/mixed.h
+// RUN: %check-in-clang -Watimport-in-framework-header -F %S/Inputs/ %t.framework/Headers/mixed.h
 
-// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -module-name Mixed -import-objc-header %S/Inputs/Mixed.framework/Headers/Mixed.h %s -typecheck -emit-objc-header-path %t/mixed-header.h
-// RUN: %FileCheck -check-prefix=CHECK -check-prefix=HEADER %s < %t/mixed-header.h
-// RUN: %check-in-clang -include %S/Inputs/Mixed.framework/Headers/Mixed.h %t/mixed-header.h
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -module-name Mixed -import-objc-header %S/Inputs/Mixed.framework/Headers/Mixed.h %s -typecheck -emit-objc-header-path %t.framework/mixed-header.h
+// RUN: %FileCheck -check-prefix=CHECK -check-prefix=HEADER %s < %t.framework/mixed-header.h
```

Comment:
Nitpick: Did you want to put this in the Headers directory?

Reply:
I don't think so, but I could be wrong. My understanding is that the first header it generates is the one that might actually be packaged in a framework, while the second is one that might be generated as a temporary file when using the framework. But I could be misunderstanding the test; I'm not too familiar with why you would or wouldn't pass `-import-underlying-module -parse-as-library` when doing `-emit-objc-header-path`.

Reply:
There's only one generated header; the first header listed is effectively a bridging header. (If you wanted to closely more emulate a mixed-source framework, you could provide a module map and use `-import-underlying-module` instead. But then only `public` things get printed.)

Oh, and the generated header is always considered part of a framework build product.

Reply:
Okay, so we're not really simulating anything very specifically here. Might as well move the header, then.